compiler/rustc_middle/src/mir/interpret/value.rs

```diff
@@ -364,13 +382,18 @@ impl<'tcx, Tag> Scalar<Tag> {
     }
 
     #[inline]
-    pub fn assert_int(self) -> ScalarInt {
+    pub fn to_int(self) -> InterpResult<'tcx, ScalarInt> {
```

Comment:
We used to have such an operation but got rid of it since code that uses it is basically always buggy (`force_int` should be used instead)... what is this used for?

Reply:
I removed it again. I think I'm able to avoid it (and it wasn't necessary for this PR)